src/Model/ApiToken.php

```diff
@@ -91,6 +94,31 @@ public static function createToken(Model $user, $ipAddress)
         return $token;
     }
 
+    /**
+     * Looks up an existing valid token for the user at the specified IP address. If none is found, it
+     * creates a new one.
+     *
+     * @param Model $user
+     * @param string $ipAddress Usually the current HTTP requester's IP, retrieved from $_SERVER[REMOTE_ADDR]
+     * @return ApiToken
+     */
+    public static function retrieveOrCreateToken(Model $user, $ipAddress)
+    {
+        try {
+            $token = self::findFirst(new AndGroup([
+                new Equals("AuthenticatedUserID", $user->UniqueIdentifier),
+                new Equals("IpAddress", $ipAddress),
+                new GreaterThan("Expires", "now", true)
+            ]));
+
+            $token->save();
```

Comment:
I think we should extend it - I'm happy tokens live for long periods if constantly in use, as long as we revoke them when a password is reset or the user is deactivated etc. Kind of like Facebook - it's token for the API seems to last indefinitely